src/network/mod.rs

```diff
@@ -0,0 +1,108 @@
+pub mod client;
+pub mod replication;
+pub mod server;
+
+use std::str::FromStr;
+
+use bevy::prelude::*;
+use bevy_egui::{
+    egui::{self},
+    EguiContexts,
+};
+use bevy_replicon::{
+    renet::RenetServer, replicon_core::NetworkChannels, server::TickPolicy, ReplicationPlugins,
+};
+
+use crate::character::player::LocalPlayerResource;
+
+use self::{
+    client::Client,
+    server::{Server, ServerPlugin},
+};
+
+pub const MAX_TICK_RATE: u16 = 30;
+pub const PROTOCOL_ID: u64 = 0;
+
+pub struct NetworkPlugin;
+
+impl Plugin for NetworkPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_plugins(
+            ReplicationPlugins
+                .build()
+                .set(bevy_replicon::server::ServerPlugin {
+                    tick_policy: TickPolicy::MaxTickRate(MAX_TICK_RATE),
+                    ..default()
+                }),
+        )
+        .insert_resource(NetworkUiState {
+            address: String::from_str("127.0.0.1:13001").unwrap(),
+        })
+        .add_plugins((replication::ReplicationPlugin, ServerPlugin))
+        .add_systems(Update, ui);
+    }
+}
+
+#[derive(Resource)]
+pub struct NetworkUiState {
```

Comment:
Maybe rename this to `MultiplayerUiState`?

Reply:
Why?

Reply:
The UI is only concerned with connecting to a multiplayer server, hence I thought it would be more appropriate. It doesn't seem to be for debugging networking issues and things like that. (I'd make that a separate window.)